### PR TITLE
identifier: allow unknown schemas

### PR DESCRIPTION
**Pull request by ntarocco**
closes https://github.com/inveniosoftware/marshmallow-utils/issues/34

### CODE REVIEW

**Review by ppanero — COMMENTED** [review]
Very nice work! 🚀 On the other hand we need to take a design decission:

- What happens if we set `fail_on_unknown=False` but then via API by mistake we submit a typo in the PID e.g. `dio` instead of `doi`... it will be accepted. Another option would be to:

"Fail on unknown if no possible schema is detected" Although we would need to trust that there are no collisions in format... What do you think?

File: marshmallow_utils/schemas/identifier.py [hunk 1]
```diff
@@ -23,40 +23,51 @@ class IdentifierSchema(Schema):
...
-
-        if self.allow_all:
-            return detected_schemes[0] if detected_schemes else None
-
-        for d in detected_schemes:
-            if d in self.allowed_schemes:
-                return d
-
-        return None
+        return detected_schemes[0] if detected_schemes else None
```
[the remark this answers is not in the document]
Comment (ntarocco) [reply marshmallow_utils/schemas/identifier.py]: I think I have fixed as discussed :)

File: marshmallow_utils/schemas/identifier.py [hunk 2]
```diff
@@ -23,40 +23,51 @@ class IdentifierSchema(Schema):
...
-        self.allow_all = allow_all
-        self.allowed_schemes = None if allow_all else allowed_schemes
-        self.required = required
+        if allowed_schemes and forbidden_schemes:
+            raise ValueError(
+                "Bad arguments allowed_schemea and forbidden_schemes: "
+                "only one allowed."
+            )
+
+        self.forbidden_schemes = forbidden_schemes or []
```
[the remark this answers is not in the document]
Comment (ntarocco) [reply marshmallow_utils/schemas/identifier.py]: Changed to `set`, thanks for the tips! :) I kept the for loop and avoid intersections to keep the code more understandable :)

**Review by ppanero — APPROVED** [review]
Very nice re-write! Ready for merge and even release (minor version due to API signature changes, right?)

**Pull request merged**